README.md
![screenshot](https://user-images.githubusercontent.com/2906365/47053390-adfc2b80-d17a-11e8-8c85-48a1ea1ce4d3.gif)

# react-blob

> React component that looks like liquid in space

[![NPM](https://img.shields.io/npm/v/react-blob.svg)](https://www.npmjs.com/package/react-blob) [![JavaScript Style Guide](https://img.shields.io/badge/code_style-standard-brightgreen.svg)](https://standardjs.com)

[Demo](https://hermanya.github.io/react-blob/)

## Install

```bash
npm install --save react-blob
```

## Usage

```tsx
import { Blob } from 'react-blob'

const Avatar = ({src, alt}) =>
  <Blob size="100px" src={src} alt={alt}/>

const FunnyLinkButton = ({href, ...props}) =>
  <a href={href}><Blob size="2em" {...props}/></a>

const BackgroundBlob = ({style, props}) =>
  <Blob size="100vh"
    style={{
        position: 'absolute',
        top: '-15%',
        right: '-15%',
        zIndex: -1,
        backgroundColor: '#21D4FD',
        color: 'white',
        opacity: 0.05,
        fontSize: '50vh',
        ...style
    }}
    {...props}
  />
```

## License

MIT © [Hermanya](https://hermanya.github.io/)


CHANGELOG.md
# Changelog

## 2.1.0
- added animation props:
  ```
  /**
   * Default is "15s"
   */
  animationDuration?: string;
  /**
   * Default is "linear"
   */
  animationTimingFunction?: "ease" | "linear" | "ease-in-out";
  /**
   * Default is undefined, but you could set it to something like "2s"
   */
  animationDelay?: string;
  /**
   * Default is "infinite"
   */
  animationIterationCount?: "infinite" | number;
  ```

## 2.0.0
- added proper TS types


src/test.ts
import { Blob } from "./";

describe("Blob", () => {
  it("is truthy", () => {
    expect(Blob).toBeTruthy();
  });
});


rollup.config.js
import typescript from "rollup-plugin-typescript2";
import commonjs from "rollup-plugin-commonjs";
import external from "rollup-plugin-peer-deps-external";
import postcss from "rollup-plugin-postcss-modules";
import resolve from "rollup-plugin-node-resolve";
import url from "rollup-plugin-url";

import pkg from "./package.json";

export default {
  input: "src/index.tsx",
  output: [
    {
      file: pkg.main,
      format: "cjs",
      exports: "named",
      sourcemap: true
    },
    {
      file: pkg.module,
      format: "es",
      exports: "named",
      sourcemap: true
    }
  ],
  plugins: [
    external(),
    postcss({
      modules: true,
      writeDefinitions: true
    }),
    url(),
    resolve(),
    typescript({
      rollupCommonJSResolveHack: true
    }),
    commonjs()
  ]
};


service-worker.js
"use strict";var precacheConfig=[["/react-blob/index.html","3386f1d7968eb12805deecc7ad824d85"],["/react-blob/static/css/main.02151954.css","b7f1244075bf66e35ef99e8b0ec4c77d"],["/react-blob/static/js/main.1cf72182.js","20a6cb8411b22d9c7478bd36705a68a8"]],cacheName="sw-precache-v3-sw-precache-webpack-plugin-"+(self.registration?self.registration.scope:""),ignoreUrlParametersMatching=[/^utm_/],addDirectoryIndex=function(e,t){var n=new URL(e);return"/"===n.pathname.slice(-1)&&(n.pathname+=t),n.toString()},cleanResponse=function(t){return t.redirected?("body"in t?Promise.resolve(t.body):t.blob()).then(function(e){return new Response(e,{headers:t.headers,status:t.status,statusText:t.statusText})}):Promise.resolve(t)},createCacheKey=function(e,t,n,r){var a=new URL(e);return r&&a.pathname.match(r)||(a.search+=(a.search?"&":"")+encodeURIComponent(t)+"="+encodeURIComponent(n)),a.toString()},isPathWhitelisted=function(e,t){if(0===e.length)return!0;var n=new URL(t).pathname;return e.some(function(e){return n.match(e)})},stripIgnoredUrlParameters=function(e,n){var t=new URL(e);return t.hash="",t.search=t.search.slice(1).split("&").map(function(e){return e.split("=")}).filter(function(t){return n.every(function(e){return!e.test(t[0])})}).map(function(e){return e.join("=")}).join("&"),t.toString()},hashParamName="_sw-precache",urlsToCacheKeys=new Map(precacheConfig.map(function(e){var t=e[0],n=e[1],r=new URL(t,self.location),a=createCacheKey(r,hashParamName,n,/\.\w{8}\./);return[r.toString(),a]}));function setOfCachedUrls(e){return e.keys().then(function(e){return e.map(function(e){return e.url})}).then(function(e){return new Set(e)})}self.addEventListener("install",function(e){e.waitUntil(caches.open(cacheName).then(function(r){return setOfCachedUrls(r).then(function(n){return Promise.all(Array.from(urlsToCacheKeys.values()).map(function(t){if(!n.has(t)){var e=new Request(t,{credentials:"same-origin"});return fetch(e).then(function(e){if(!e.ok)throw new Error("Request for "+t+" returned a response with status "+e.status);return cleanResponse(e).then(function(e){return r.put(t,e)})})}}))})}).then(function(){return self.skipWaiting()}))}),self.addEventListener("activate",function(e){var n=new Set(urlsToCacheKeys.values());e.waitUntil(caches.open(cacheName).then(function(t){return t.keys().then(function(e){return Promise.all(e.map(function(e){if(!n.has(e.url))return t.delete(e)}))})}).then(function(){return self.clients.claim()}))}),self.addEventListener("fetch",function(t){if("GET"===t.request.method){var e,n=stripIgnoredUrlParameters(t.request.url,ignoreUrlParametersMatching),r="index.html";(e=urlsToCacheKeys.has(n))||(n=addDirectoryIndex(n,r),e=urlsToCacheKeys.has(n));var a="/react-blob/index.html";!e&&"navigate"===t.request.mode&&isPathWhitelisted(["^(?!\\/__).*"],t.request.url)&&(n=new URL(a,self.location).toString(),e=urlsToCacheKeys.has(n)),e&&t.respondWith(caches.open(cacheName).then(function(e){return e.match(urlsToCacheKeys.get(n)).then(function(e){if(e)return e;throw Error("The cached response that was expected is missing.")})}).catch(function(e){return console.warn('Couldn\'t serve response for "%s" from cache: %O',t.request.url,e),fetch(t.request)}))}});

example/src/App.js
import React, { Component } from "react";

import { Blob } from "react-blob";

const codeExample = `import * as React from 'react'
import { Blob } from 'react-blob'

const Avatar = ({src, alt}) =>
  <Blob size="100px" src={src} alt={alt}/>

const FunnyLinkButton = ({href, ...props}) =>
  <a href={href}><Blob size="2em" {...props}/></a>

const BackgroundBlob = ({style, props}) =>
  <Blob size="100vh"
    style={{
        position: 'absolute',
        top: '-15%',
        right: '-15%',
        zIndex: -1,
        backgroundColor: '#21D4FD',
        color: 'white',
        opacity: 0.05,
        fontSize: '50vh',
        ...style
    }}
    animationDuration="60s"
    {...props}
  />`;

export default class App extends Component {
  render() {
    return (
      <div className="overflow-hidden position-relative">
        <div className="small-container px-3 px-md-0 mx-auto py-5">
          <div className="d-flex align-items-center justify-content-center">
            <Blob
              size="5em"
              style={{ fontSize: "2.5em" }}
              className="bg-secondary text-white"
            >
              react-blob
            </Blob>
            <a href="https://github.com/hermanya/react-blob">
              <Blob size="2em" className="bg-dark text-white">
                <i className="fab fa-github"></i>
              </Blob>
            </a>
          </div>
          <section className="mb-4 text-center animated bounceInUp delay-01s">
            <h1 className="display-4 text-muted">
              React component that looks like <strong>liquid in space</strong>
            </h1>
          </section>
          <div className="row">
            <section className="col animated bounceInUp delay-02s">
              <h2 className="mb-4">Install</h2>
              <pre className="bg-dark text-success p-3 rounded">
                <code>npm install --save react-blob</code>
              </pre>
              <a href="https://www.npmjs.com/package/react-blob">
                <img
                  src="https://img.shields.io/npm/v/react-blob.svg"
                  alt="npm version"
                  className="mr-2"
                />
              </a>
              <img
                src="https://img.shields.io/badge/code_style-standard-brightgreen.svg"
                alt="code style"
              />
              <div className="d-flex align-items-center mt-2">
                MIT ©
                <a href="https://hermanya.github.io" className="ml-1">
                  Herman Starikov
                </a>
                <Blob
                  size="2em"
                  className="d-inline-block ml-2"
                  src="https://s3-us-west-2.amazonaws.com/s.cdpn.io/1589206/profile/profile-512.jpg"
                />
              </div>
            </section>
            <section className="col animated bounceInUp delay-03s">
              <h2 className="mb-4">Usage</h2>
              <pre className="bg-dark text-success p-3 rounded">
                <code>{codeExample}</code>
              </pre>
            </section>
          </div>
        </div>
        <Blob
          size="100vh"
          style={{
            position: "absolute",
            top: "-15%",
            right: "-15%",
            zIndex: -1,
            backgroundColor: "#21D4FD",
            color: "white",
            opacity: 0.15,
            fontSize: "50vh"
          }}
          animationDuration="60s"
        >
          <i className="fab fa-react"></i>
        </Blob>
      </div>
    );
  }
}
